android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt

```diff
@@ -140,14 +140,20 @@ constructor(
 
   suspend fun <R : Resource> addOrUpdate(addMandatoryTags: Boolean = true, resource: R) {
     return withContext(dispatcherProvider.io()) {
-      resource.updateLastUpdated()
-      try {
-        fhirEngine.get(resource.resourceType, resource.logicalId).run {
-          fhirEngine.update(updateFrom(resource))
-        }
-      } catch (resourceNotFoundException: ResourceNotFoundException) {
-        create(addMandatoryTags, resource)
+      if (resource.hasMeta() && resource.meta.hasLastUpdated()) {
+        resource.updateLastUpdated()
       }
+      kotlin
+        .runCatching {
```

Comment:
Internally this calls `try-catch` block actually? I'd prefer the `try-catch ` to this as we are catching a specific exception. The `runCatching` returns a `Throwable which means you have to explicitly check that the exception instance matches the one you are interested in . I think since we are sure at this point the only exception to be returned is `ResourceNotFoundException` then the `try-catch` version is better.